### DISCUSSION TITLE
Priority field is removed from message after syslog parsed

### DISCUSSION THREAD

**Discussion by chetaksirsat** [fluent/fluentd#3804] [Q&A]
My fluentd collector today parses syslogs according to our config below but in the process removes the priority field from the message making it a non-compliant RFC format.

**BEFORE: Original RFC5424 compliant format message**
**<99>** 1 2022-06-15T11:11:38.313536+00:00 vc vapi-endpoint-access - - - 2022-06-15T11:11:38.313Z | jetty-default-497413 | Invoking operation get from service com.vmware.cis.tagging.tag with id bbfe266d-d31e-4374-9df5-ab261d2bf074

**AFTER: Parsed syslog message using Fluentd**
1 2022-06-15T11:11:38.313536+00:00 vc vapi-endpoint-access - - - 2022-06-15T11:11:38.313Z | jetty-default-497413 | Invoking operation get from service com.vmware.cis.tagging.tag with id bbfe266d-d31e-4374-9df5-ab261d2bf074

Is there a way to maintain the priority field **<99>** in the message itself? 

Below is the fluentd config - 

```
<source>
  @type syslog

  <parse>
   @type multi_format

   <pattern>
    format regexp
    expression /(?<external>.{0})(?<message>.*)/
   </pattern>
  </parse>

  @log_level info
  format none
  port 514
  tag selector

  severity_key severity
  facility_key facility
</source>
```

**Comment by subigre** [comment]
Hello @chetaksirsat,

I know this topic is quiet old but I just found it and I have exactly the same requirements.
Did you find any solution to keep the `PRI` in the message.

Regards

**Comment by daipom** [comment]
@chetaksirsat @subigre 
`in_syslog` always parses `PRI` and set the severity and the facility to `severity_key`, `facility_key`.

If you need to keep `PRI` in the message, you can use `in_tcp` or `in_udp` instead of `in_syslog`.
